framework/src/timeintegrators/ActuallyExplicitEuler.C

```diff
@@ -0,0 +1,236 @@
+  // have had solveSetup() called, and sets the default solver
+  // parameters for PETSc.
+  _fe_problem.initPetscOutput();
+
+  auto & es = _fe_problem.es();
+
+  auto & nonlinear_system = _fe_problem.getNonlinearSystemBase();
+
+  auto & libmesh_system = dynamic_cast<NonlinearImplicitSystem &>(nonlinear_system.system());
+
+  auto & mass_matrix = *libmesh_system.matrix;
+
+  _current_time = _fe_problem.time();
+
+  // Set time back so that we're evaluating the interior residual at the old time
+  _fe_problem.time() = _fe_problem.timeOld();
+
+  libmesh_system.update();
+
+  // Must compute the residual first
+  _explicit_residual.zero();
+  _fe_problem.computeResidual(*libmesh_system.current_local_solution, _explicit_residual);
+
+  // The residual is on the RHS
+  _explicit_residual *= -1.;
+
+  // Compute the mass matrix
+  _fe_problem.computeJacobianTag(*libmesh_system.current_local_solution, mass_matrix, _Ke_time_tag);
+
+  // Still testing whether leaving the old update is a good idea or not
+  // _explicit_euler_update = 0;
+
+  switch (_solve_type)
+  {
+    case CONSISTENT:
+      _linear_solver->solve(mass_matrix,
+                            _explicit_euler_update,
+                            _explicit_residual,
+                            es.parameters.get<Real>("linear solver tolerance"),
+                            es.parameters.get<unsigned int>("linear solver maximum iterations"));
+
+      break;
+
+    case LUMPED:
+      // Computes the sum of each row (lumping)
+      // Note: This is actually how PETSc does it
+      // It's not "perfectly optimal" - but it will be fast (and universal)
+      mass_matrix.vector_mult(_mass_matrix_diag, *_ones);
+
+      // "Invert" the diagonal mass matrix
+      _mass_matrix_diag.reciprocal();
+
+      // Multiply the inversion by the RHS
+      _explicit_euler_update.pointwise_mult(_mass_matrix_diag, _explicit_residual);
+      break;
+
+    case LUMP_PRECONDITIONED:
+      mass_matrix.vector_mult(_mass_matrix_diag, *_ones);
+      _mass_matrix_diag.reciprocal();
+
+      _linear_solver->solve(mass_matrix,
+                            _explicit_euler_update,
+                            _explicit_residual,
+                            es.parameters.get<Real>("linear solver tolerance"),
+                            es.parameters.get<unsigned int>("linear solver maximum iterations"));
+      break;
+
+    default:
+      mooseError("Unknown solve_type in ActuallyExplicitEuler ");
+  }
+
+  *libmesh_system.solution = nonlinear_system.solutionOld();
+  *libmesh_system.solution += _explicit_euler_update;
+
+  // Enforce contraints on the solution
+  DofMap & dof_map = libmesh_system.get_dof_map();
+  dof_map.enforce_constraints_exactly(libmesh_system, libmesh_system.solution.get());
+
+  libmesh_system.update();
+
+  nonlinear_system.setSolution(*libmesh_system.current_local_solution);
+
+  libmesh_system.nonlinear_solver->converged = true;
```

Comment:
If we skip the PETSc solve like in lumped case and the RHS evaluation produces a NaN, converged should be `false`.

Reply:
Yeah - the linear solve can fail too.  I'll put some better checks for convergence.